lib/src/main/java/com/cloudbees/groovy/cps/impl/CpsClosureDef.java

```diff
@@ -34,6 +35,19 @@ Next invoke(Env caller, SourceLocation loc, Object receiver, List<?> args, Conti
 
         // TODO: who handles 'it' ?
 
+        // 'e' will be a DepthTrackingEnv for all pipelines begun with this version of the library
+        // but if we upgrade groovy-cps and resume older pipelines, they won't be.
```

Comment:
Is that actually true? `DepthTrackingEnv` is just a marker interface. Existing instances of classes which retroactively implement it will be `instanceof DepthTrackingEnv` after a restart.

I think what you meant to say is that `getDepth()` would be 0 in such cases, since the field would be null after deserialization.

Reply:
I'm honestly not 100% sure if we'll have a path that provides an Env that isn't a DepthTrackingEnv -- but I'm also cautious about things like this.  We can revise the comment if you like.

Reply:
You know what,  I don't know what I was thinking.  We don't need to worry about this at all, fixing it.